linkerd/protocol/h2/src/main/scala/io/buoyant/linkerd/protocol/h2/HeaderPathIdentifier.scala

```diff
@@ -0,0 +1,108 @@
+package io.buoyant.linkerd
+package protocol
+package h2
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.twitter.finagle.{Dtab, Path, Stack}
+import com.twitter.finagle.buoyant.Dst
+import com.twitter.finagle.buoyant.h2.{Headers => H2Headers, LinkerdHeaders, Request}
+import com.twitter.util.Future
+import io.buoyant.router.H2
+import io.buoyant.router.RoutingFactory._
+
+class HeaderPathIdentifier(
+  header: String,
+  segments: Option[Int],
+  pfx: Path,
+  baseDtab: () => Dtab
+) extends Identifier[Request] {
+  import HeaderPathIdentifier._
+
+  private[this] val identifyPath: Request => Future[RequestIdentification[Request]] =
+    segments match {
+      case Some(segments) if segments < 1 =>
+        throw new IllegalArgumentException("segments must be >= 1")
+
+      case Some(segments) =>
+        lazy val tooShort = {
+          val msg = s"Path must have at least $segments segment(s)"
+          Future.value(new UnidentifiedRequest[Request](msg))
+        }
+        (req: Request) => reqPath(req) match {
+          case p if p.size < segments => tooShort
+          case p => identified(p, req)
+        }
+
+      case None =>
+        (req: Request) => identified(reqPath(req), req)
+    }
+
+  private[this] def reqPath(req: Request): Path =
```

Comment:
Yeah, sure. I'm skeptical about that feature in general. But if we want to add it, it would go here.